azure/services/async/async.go

```diff
@@ -230,3 +244,9 @@ func getRetryAfterFromError(err error) time.Duration {
 	}
 	return ret
 }
+
+// ContextDeadlineExceededOrCanceled checks if it's an intermittent context deadline
+// exceeded or canceled error.
+func ContextDeadlineExceededOrCanceled(ctx context.Context) bool {
+	return errors.Is(ctx.Err(), context.DeadlineExceeded) || errors.Is(ctx.Err(), context.Canceled)
```

Comment:
I am wondering if `context.Canceled` qualifies as an _transient error_?
I understand `context.Canceled` as `abandon work`. So, shouldn't we be bubbling up the error or requeuing the request upon `context.Canceled`?

Reply:
Sorry my bad. I looked it up, and `azure.WithTransientError()` bubbles up the error as transient and also requeues with default timeout unless other time is specified in the error. https://github.com/nawazkh/cluster-api-provider-azure/blob/f8ad961ef46fe856eda6d96056751e8b7e4a068f/azure/services/async/async.go#L208

Reply:
can we move this helper func to https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/0837cd3019526060dc334b9b1e9cb82d94f44e64/azure/errors.go and add a small unit test for it?

Reply:
Sure. I had added a review comment re this myself to keep it in azure/errors.go or not ( got outdated after I pushed the changes )
I decided to keep it here : 
- Pushing it to azure/errors.go will mean an import of ctx 
- I thought it is worth to keep it here until a situation arises when we need to use this function at other place which will give more info on what should be the right home for this. 

But also, I do not see a problem to push it to azure/errors.go and I will do that for now.